main.tf

```
@@ -130,10 +140,12 @@ resource "aws_route53_record" "alb" {
     zone_id                = "${module.alb.load_balancer_zone_id}"
     evaluate_target_health = true
   }
+
+  count = "${data.aws_partition.current.partition == "aws" ? 1 : 0}"
 }
 
 resource "aws_route53_record" "alb-subdomain" {
-  count   = "${ var.enable_subdomains ? 1 : 0 }"
+  count   = "${var.enable_subdomains == true ? "${data.aws_partition.current.partition == "aws" ? 1 : 0}" : 0 }"
```

As far as we don't support subdomains in china (no matter on module input parameters):
```suggestion
  count   = "${data.aws_partition.current.partition == "aws" ? "${var.enable_subdomains == true ? 1 : 0}" : 0 }"
```

---

fixed